pkg/pebinaryexecutor/pebinary.go

```diff
@@ -0,0 +1,268 @@
+		logrus.Infof("WinDSR support is enabled")
+		extraArgs["feature-gates"] = addFeatureGate(extraArgs["feature-gates"], "WinDSR=true")
+		extraArgs["enable-dsr"] = "true"
+	}
+
+	for range time.Tick(time.Second * 5) {
+		endpoint, err := hcsshim.GetHNSEndpointByName("Calico_ep")
+		if err != nil {
+			logrus.WithError(err).Warning("can't find %s, retrying", "Calico_ep")
+			continue
+		}
+		extraArgs["source-vip"] = endpoint.IPAddress.String()
+		break
+	}
+
+	logrus.Infof("Deleting HNS policies before kube proxy starts.")
+	policies, _ := hcsshim.HNSListPolicyListRequest()
+	for _, policy := range policies {
+		policy.Delete()
+	}
+
+	args = append(getArgs(extraArgs), args...)
+	logrus.Infof("Running RKE2 Kubeproxy %s", args)
+	go func() {
+		for {
+			cmd := exec.Command(filepath.Join(p.DataDir, "bin", "kube-proxy.exe"), args...)
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+			err := cmd.Run()
+			logrus.Errorf("Kubeproxy exited: %v", err)
+			time.Sleep(5 * time.Second)
+		}
+	}()
+
+	return nil
+}
+
+// APIServer isn't supported in the binary executor.
+func (p *PEBinary) APIServer(ctx context.Context, etcdReady <-chan struct{}, args []string) (authenticator.Request, http.Handler, error) {
+	panic("api server is unsupported on windows")
+}
+
+// Scheduler isn't supported in the binary executor.
+func (p *PEBinary) Scheduler(apiReady <-chan struct{}, args []string) error {
+	panic("scheduler is unsupported on windows")
+}
+
+// ControllerManager isn't supported in the binary executor.
+func (p *PEBinary) ControllerManager(apiReady <-chan struct{}, args []string) error {
+	panic("controller manager is unsupported on windows")
+}
+
+// CurrentETCDOptions isn't supported in the binary executor.
+func (p *PEBinary) CurrentETCDOptions() (opts executor.InitialOptions, err error) {
+	panic("etcd options are unsupported on windows")
+}
+
+// CloudControllerManager isn't supported in the binary executor.
+func (p *PEBinary) CloudControllerManager(ccmRBACReady <-chan struct{}, args []string) error {
+	panic("cloud controller manager is unsupported on windows.")
+}
+
+// ETCD isn't supported in the binary executor.
+func (p *PEBinary) ETCD(args executor.ETCDConfig) error {
+	panic("etcd is unsupported on windows")
+}
+
+// addFeatureGate adds a feature gate with the correct syntax.
+func addFeatureGate(current, new string) string {
+	if current == "" {
+		return new
+	}
+	return current + "," + new
+}
+
+// getArgs concerts a map to the correct args list format.
+func getArgs(argsMap map[string]string) []string {
+	args := make([]string, len(argsMap))
+	for arg, value := range argsMap {
+		cmd := fmt.Sprintf("--%s=%s", arg, value)
+		args = append(args, cmd)
+	}
+	sort.Strings(args)
+	return args
+}
+
+func getCniType(restConfig *rest.Config) (string, error) {
+	hc, err := helm.NewFactoryFromConfig(restConfig)
+	if err != nil {
+		return "", err
+	}
+	hl, err := hc.Helm().V1().HelmChart().List(metav1.NamespaceSystem, metav1.ListOptions{})
```

Comment:
You should be able to see what's in the cli --disable list, instead of poking at the HelmChart resources.

If for some reason you need to check for the HelmChart resource, just try to Get() the `rke2-calico` HelmChart rather than listing and seeing if it's in the list.

Reply:
I am listing because we plan to support multiple CNIs. We will need see which one is installed to determine which implementation to use.